types/plotly.js/index.d.ts

```diff
@@ -301,8 +301,16 @@ export interface PolarLayout {
     uirevision: string | number;
 }
 
+export interface PlotlyDataLayoutConfig {
+    data: Data[];
+    layout?: Partial<Layout>;
+    config?: Partial<Config>;
+}
+
 export type Root = string | HTMLElement;
 
+export type RootOrData = string | HTMLElement | PlotlyDataLayoutConfig;
```

Comment:
```suggestion
export type RootOrData = Root | PlotlyDataLayoutConfig;
```
would made more sense for reader (I know TS does not care here)